chia/consensus/blockchain.py

```diff
@@ -336,14 +337,14 @@ async def add_block(
         if extending_main_chain:
             fork_info.reset(block.height - 1, block.prev_header_hash)
 
-        block_rec = await self.get_block_record_from_db(header_hash)
-        if block_rec is not None:
+        block_rec_from_db = await self.get_block_record_from_db(header_hash)
```

Comment:
we check the db anyway since we might be passing a converted record from a staged chain in a sync

Reply:
this looks like it ought to be a comment in the code